lib/ladybug/energy_model/aperture.rb

```diff
@@ -0,0 +1,90 @@
+# may be used to endorse or promote products derived from this software without
+# specific prior written permission from the respective party.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE
+# UNITED STATES GOVERNMENT, OR THE UNITED STATES DEPARTMENT OF ENERGY, NOR ANY OF
+# THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+# OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *******************************************************************************
+
+require 'ladybug/energy_model/model_object'
+
+require 'json-schema'
+require 'json'
+require 'openstudio'
+
+module Ladybug
+  module EnergyModel
+    class Aperture < ModelObject
+      attr_reader :errors, :warnings
+
+      def initialize(hash)
+        super(hash)
+
+        raise "Incorrect model type '#{@type}'" unless @type == 'Aperture'
+      end
+
+      def defaults
+        result = {}
+        result[:type] = @@schema[:definitions][:Aperture][:properties][:type][:enum]
+        result
+      end
+
+      def find_existing_openstudio_object(openstudio_model)
+        object = openstudio_model.getSubSurfaceByName(@hash[:name])
+        return object.get if object.is_initialized
+        nil
+      end
+
+      def create_openstudio_object(openstudio_model)
+        openstudio_vertices = OpenStudio::Point3dVector.new
+        @hash[:vertices].each do |vertex|
+          openstudio_vertices << OpenStudio::Point3d.new(vertex[:x], vertex[:y], vertex[:z])
+        end
+
+        construction = nil
+        if @hash[:energy_construction_opaque]
+          construction_opaque_name = @hash[:energy_construction_opaque][:name]
+          construction_opaque = openstudio_model.getConstructionByName(construction_opaque_name)
+          unless construction_opaque.empty?
+            construction = construction_opaque.get
+          end
+        else
+          construction_transparent_name = @hash[:energy_construction_transparent][:name]
+          construction_transparent = openstudio_model.getConstructionByName(construction_transparent_name)
+          unless construction_transparent.empty?
+            construction = construction_transparent.get
+          end
+        end
+
+        openstudio_subsurface = OpenStudio::Model::SubSurface.new(openstudio_vertices, openstudio_model)
+        openstudio_subsurface.setName(@hash[:name])
```

Comment:
@chriswmackey we could import id, something like 

`openstudio_subsurface.additionalProperties.setFeature('LadybugId', @hash[:id])`

Reply:
Seems like should we do this on all objects?  We can add code to build an index of OpenStudio objects by Ladybug Id.